### PR TITLE
[XLA:GPU] Use cuGraphInstantiateWithParams to instantiate cuda graphs for better debug.

### PR DESCRIPTION
**Pull request by shawnwang18**
This PR uses cuGraphInstantiateWithParms instead of cuGraphInstantiate to instantiate cuda graph executors, so current command_buffer_cmd_test and command_buffer_thunk_test should cover the changes in this PR.

### CODE REVIEW

File: xla/stream_executor/cuda/cuda_driver.cc [hunk 1]
```diff
@@ -1221,75 +1253,75 @@ absl::Status GpuDriver::LoadPtx(GpuContext* context, const char* ptx_contents,
...
-
-    absl::Status status;
-    {
-      // TODO(leary) Need to see if NVIDIA can expunge the leakiness in their
-      // module loading: see http://b/13248943
-      absl::LeakCheckDisabler disabler;
-      status = cuda::ToStatus(cuModuleLoadDataEx(
-          module, ptx_data, TF_ARRAYSIZE(options), options, option_values));
-    }
+  GetDriverExecutor()->Schedule(
```
Comment (ezhulenev) [inline xla/stream_executor/cuda/cuda_driver.cc:1256]: This looks just a formatting change? Or I can't find a real one? If it's just formatting can you please revert it, I'll submit NFC formatting later separately

File: xla/stream_executor/gpu/gpu_types.h [hunk 2]
```diff
@@ -18,6 +18,7 @@ limitations under the License.
 #ifndef XLA_STREAM_EXECUTOR_GPU_GPU_TYPES_H_
 #define XLA_STREAM_EXECUTOR_GPU_GPU_TYPES_H_
 
+#include <string>
 #if TENSORFLOW_USE_SYCL
```
Comment (ezhulenev) [inline xla/stream_executor/gpu/gpu_types.h:22]: nit: style requires a space between std headers and everything else

Comment (shawnwang18) [reply xla/stream_executor/gpu/gpu_types.h:23]: fixed

File: xla/stream_executor/cuda/cuda_driver.cc (hunk 1 above)
Comment (shawnwang18) [reply xla/stream_executor/cuda/cuda_driver.cc]: Revert the change here, should be due to clicking the whole file format button from vscode.

**Review by ezhulenev — APPROVED** [review]

File: xla/stream_executor/gpu/gpu_driver.h [hunk 3]
```diff
@@ -346,6 +346,49 @@ class GpuDriver {
                                        GpuGraphHandle graph,
                                        const GraphInstantiateFlags& flags);
 
+  // Graph instantiation parameters
+  struct GraphInstantiateParams {
+    GraphInstantiateFlags flags;
+    GpuGraphNodeHandle err_node_out = NULL;
+    GpuStreamHandle upload_stream = NULL;
+#if CUDA_VERSION >= 12000
```
Comment (penpornk) [inline xla/stream_executor/gpu/gpu_driver.h:354]: Could you please add `#include "third_party/gpus/cuda/include/cuda.h"` for `CUDA_VERSION`?

**Review by penpornk — CHANGES_REQUESTED** [review]

File: xla/stream_executor/gpu/gpu_driver.h (hunk 3 above)
Comment (shawnwang18) [reply xla/stream_executor/gpu/gpu_driver.h:355]: Fixed, could you help make the merge?

**Review by cheshire — APPROVED** [review]

**Review by ezhulenev — APPROVED** [review]

File: xla/stream_executor/gpu/gpu_driver.h (hunk 3 above)
Comment (akuegel) [reply xla/stream_executor/gpu/gpu_driver.h:357]: @shawnwang18 Eugene is right, gpu_driver.h should be platform agnostic, so no Nvidia specific #ifdef is allowed here.

**Pull request closed without merging**